File: Miscellaneous Questions/Algorithms in C#/bubbleSort.cs
using System;

namespace BubbleSortDemo
{
    class Program
    {
        static void Main(string[] args)
        {
            // create an array of unsorted integers
            int[] nums = {5, 3, 8, 2, 1, 4};

            // print out the unsorted array
            Console.WriteLine("Unsorted Array:");
            foreach (int n in nums)
            {
                Console.Write(n + " ");
            }
            Console.WriteLine();

            // sort the array using bubble sort
            BubbleSort(nums);

            // print out the sorted array
            Console.WriteLine("Sorted Array:");
            foreach (int n in nums)
            {
                Console.Write(n + " ");
            }
            Console.WriteLine();
        }

        static void BubbleSort(int[] arr)
        {
            // keep track of whether or not an element was swapped
            bool swapped;

            do
            {
                // assume no elements were swapped
                swapped = false;


                for (int i = 0; i < arr.Length - 1; i++)
                {
                    // if two adjacent elements are out of order
                    if (arr[i] > arr[i + 1])
                    {
                        // swap them
                        int temp = arr[i];
                        arr[i] = arr[i + 1];
                        arr[i + 1] = temp;

                        // indicate that an element was swapped
                        swapped = true;
                    }
                }
            } while (swapped);
        }
    }
}

File: Miscellaneous-Questions/Algorithms in C++/bubbleSort.cpp
#include <iostream> 
using namespace std; 

// Function to swap two elements A[i] and A[j] 
void swap(int A[], int i, int j) 
{ 
	int temp = A[i]; 
	A[i] = A[j]; 
	A[j] = temp; 
} 

// Function to implement bubble sort 
void bubbleSort(int A[], int n) 
{ 
	int i, j; 
	for (i = 0; i < n-1; i++)	 
	
	// Last i elements are already in place 
	for (j = 0; j < n-i-1; j++) 
		if (A[j] > A[j+1]) 
			swap(A, j, j+1); 
} 

// Function to print an array 
void printArray(int A[], int size) 
{ 
	int i; 
	for (i=0; i < size; i++) 
	cout << A[i] << " "; 
	cout << endl; 
} 

// Driver program to test above functions 
int main() 
{ 
	int A[] = {64, 34, 25, 12, 22, 11, 90}; 
	int n = sizeof(A)/sizeof(A[0]); 
	bubbleSort(A, n); 
	cout<<"Sorted array: \n"; 
	printArray(A, n); 
	return 0; 
}